test/cbmc/sources/core_sntp_cbmc_state.c

```diff
@@ -0,0 +1,115 @@
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file core_sntp_cbmc_state.c
+ * @brief Implements the functions defined in core_sntp_cbmc_state.h.
+ */
+#include <stdint.h>
+#include <stdlib.h>
+#include "core_sntp_client.h"
+#include "core_sntp_cbmc_state.h"
+#include "core_sntp_stubs.h"
+
+SntpContext_t * allocateCoreSntpContext( SntpContext_t * pContext )
+{
+    SntpServerInfo_t * pTimeServers;
+    size_t currentServerIndex;
+    size_t numOfServers;
+    uint32_t serverResponseTimeoutMs;
+    uint8_t * pNetworkBuffer;
+    size_t bufferSize;
+    SntpResolveDns_t resolveDnsFunc;
+    SntpGetTime_t getTimeFunc;
+    SntpSetTime_t setTimeFunc;
+    UdpTransportInterface_t * pNetworkIntf;
+    SntpAuthenticationInterface_t * pAuthIntf;
+    SntpStatus_t sntpStatus = SntpSuccess;
+
+    if( pContext == NULL )
+    {
+        pContext = malloc( sizeof( SntpContext_t ) );
+    }
+
+    __CPROVER_assume( numOfServers < CBMC_MAX_OBJECT_SIZE );
+    __CPROVER_assume( serverResponseTimeoutMs < CBMC_MAX_OBJECT_SIZE );
+    __CPROVER_assume( currentServerIndex < CBMC_MAX_OBJECT_SIZE );
+
+    if( numOfServers == 0 )
+    {
+        pTimeServers = NULL;
+    }
+    else
+    {
+        pTimeServers = malloc( numOfServers * sizeof( SntpServerInfo_t ) );
+    }
+
+    if( pTimeServers != NULL )
+    {
+        __CPROVER_assume( pTimeServers->serverNameLen < CBMC_MAX_OBJECT_SIZE );
+        __CPROVER_assume( pTimeServers->port < CBMC_MAX_OBJECT_SIZE );
+        pTimeServers->pServerName = malloc( pTimeServers->serverNameLen );
+    }
+
+    __CPROVER_assume( bufferSize < CBMC_MAX_OBJECT_SIZE );
+    pNetworkBuffer = malloc( bufferSize );
+
+    pNetworkIntf = malloc( sizeof( UdpTransportInterface_t ) );
+
+    if( pNetworkIntf != NULL )
+    {
+        pNetworkIntf->pUserContext = malloc( sizeof( NetworkContext_t ) );
+        pNetworkIntf->sendTo = NetworkInterfaceSendStub;
+        pNetworkIntf->recvFrom = NetworkInterfaceReceiveStub;
+    }
+
+    pAuthIntf = malloc( sizeof( SntpAuthenticationInterface_t ) );
+
+    if( pAuthIntf != NULL )
+    {
+        pAuthIntf->pAuthContext = malloc( sizeof( SntpAuthContext_t ) );
+        pAuthIntf->generateClientAuth = GenerateClientAuthStub;
```

Comment:
You should set the `validateServerAuth` member also as that will be required for the `Sntp_ReceiveTimeResponse` proof

Reply:
Yeah, I plan to implement it as a part of Sntp_ReceiveTimeResponse proof, not made it part of this PR.